docs/platform-support.md

```diff
@@ -0,0 +1,44 @@
+# Platform Support
```

Comment:
IMO we should define what the different platforms are. To that end I think we can just say they are a pair consisting of an operating system family together with a processor architecture family as defined by the Go team through the GOOS and GOARCH tags. We can also comment on this area as to what operating system versions we support (specifically, I would say as a library we support all OS versions officially supported by Go on their supported versions, and as a binary all the ones on the latest version of Go).

cc @rmfitzpatrick on this, please comment if you think the OS versions supported should be different.

Reply:
Added definition of platform in this section. With respect to a generic support statement for OS and binaries need to discuss further, as it is not clear to me yet how this relates to test strategy and association with support tiers.

Reply:
> it is not clear to me yet how this relates to test strategy and association with support tiers

My point is just that if we say we support Linux/Windows, it's unclear what versions of that OS family we support and we should say that as part of the definition of each platform.

Reply:
Added statement about supported OS versions based on Go support.